api/search.go

```diff
@@ -57,7 +57,7 @@ func (api *SearchAPI) getSearch(w http.ResponseWriter, r *http.Request) {
 
 	var authToken string
 
-	if r.Header.Get(internalTokenHeader) == api.internalToken {
+	if hasAuth == true {
```

Comment:
This shouldn't be dependent on if it is run in publishing or web. We should be checking if the auth token is valid to decide whether the user is allowed to preview an unpublished document or not; it shouldn't have anything to do with whether the application is running in publishing or web.

Also hasAuth is always false (see comment above), you can access the flag by using `api.privateAuth.HasPrivateEndpoints`